wagtail/fields.py

```diff
@@ -212,8 +212,18 @@ def get_prep_value(self, value):
             )
         else:
             # When querying with JSONField features, the rhs might not be a StreamValue.
+            # Note: when Django 4.2 is the minimum supported version, this can be removed
+            # as the serialisation is handled in get_db_prep_value instead.
```

Comment:
We still need to keep this for now as in Django < 4.2, the `DataContains` and `ContainedBy` lookups do not inherit from `FieldGetDbPrepValueMixin` yet.

When we only support Django 4.2 and up, this can be removed and we can revert to the original logic pre-`use_json_field` by replacing the `elif` above with `else`:

https://github.com/wagtail/wagtail/blob/cbe9dc729b7cc92229b535c9bedd719b1a12c9e8/wagtail/core/fields.py#L137-L145